### ISSUE TITLE
tf.keras model.fit(): enormous difference between train loss and val loss on the same data

### ISSUE THREAD

**Issue by dmus** [tensorflow/tensorflow#36446]
Tensorflow version 2.1

See the colab notebook to reproduce the issue: https://drive.google.com/file/d/1Fvc6G_9v5mek015cai7qYT6HoY-fLkzk/view?usp=sharing

When the training loss goes down the val_loss does not change, although this is exactly the same data.

Train on 2 samples, validate on 2 samples
Epoch 1/30
2/2 [==============================] - 3s 2s/sample - loss: 0.4630 - val_loss: 302.4763
Epoch 2/30
2/2 [==============================] - 1s 457ms/sample - loss: 0.8565 - val_loss: 496.9578
Epoch 3/30
2/2 [==============================] - 1s 457ms/sample - loss: 0.7886 - val_loss: 1050.9148
Epoch 4/30
2/2 [==============================] - 1s 450ms/sample - loss: 0.1080 - val_loss: 744.4895
Epoch 5/30
2/2 [==============================] - 1s 474ms/sample - loss: 0.1144 - val_loss: 1353.2678
Epoch 6/30
2/2 [==============================] - 1s 465ms/sample - loss: 0.0402 - val_loss: 3237.9683
Epoch 7/30
2/2 [==============================] - 1s 465ms/sample - loss: 0.0635 - val_loss: 3946.7822
Epoch 8/30
2/2 [==============================] - 1s 470ms/sample - loss: 0.0355 - val_loss: 4054.5461
Epoch 9/30
2/2 [==============================] - 1s 462ms/sample - loss: 0.0345 - val_loss: 4991.5400

How is this possible?
The code is pretty straightforward:

```
import tensorflow as tf
import numpy as np
base_model = tf.keras.applications.ResNet50V2(input_shape=(180, 320, 3), weights=None, include_top=False)
x = tf.keras.layers.GlobalAveragePooling2D()(base_model.output)
output = tf.keras.layers.Dense(8)(x)
model = tf.keras.models.Model(inputs=base_model.input, outputs=output)
model.compile(optimizer='adam', loss='mse')
data = np.random.rand(2, 180, 320, 3)
labels = np.random.rand(2, 8)

model.fit(data, labels, validation_data=(data,labels), batch_size=2, epochs=30)
```

There are known issues with keras and batch normalization (see for example https://github.com/keras-team/keras/issues/6977). This is probably related, but I don't see directly how. What do I have to change to make this working as expected? Is this something in the included package https://github.com/qubvel/classification_models or where to solve it?

Edit: the behaviour of batch normalization is changed as of TF 2.0, so the other issues might not be related, see https://www.tensorflow.org/api_docs/python/tf/keras/layers/BatchNormalization

**Comment by ravikyram** [comment]
@dmus 

Can you please provide the access to colab link to reproduce the issue in our environment.Thanks!

**Comment by dmus** [comment]
Updated colab notebook link

**Comment by ravikyram** [comment]
@dmus 

I have tried in colab with TF version 2.2.0-dev20200205 and i am not seeing any issue. Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/053d84cd47d7653c362ac6d7bc19ab04/untitled623.ipynb). Is this the expected behavior? Thanks!

**Comment by dmus** [comment]
@ravikyram Thanks for having a look. I think indeed there is no bug, but this is the expected behaviour of BatchNormalization layers. It takes time before the batch statistics are similar to the statistics used in a batch, explaining the big difference between train and val loss.
Eventually this can be speed up by lowering the momentum property of the BatchNormalization, default is 0.99

**Comment by ccg1415** [comment]
I'm facing the same problem. Transfer learning with Mobilenetv2, all layers from mobilenet are set as _not trainable_ (eg layer.trainable=False), rest of the network is similar to what is in issue description. Same data (tfrecord dataset) are passed in as train data and also as validation data. Training stats looks good. loss goes quickly to zero, accuracy goes quickly to 1.

So far, all as it should be. Validation stats on the other hand are broken. Validation loss goes up and validation accuracy doesn't change. Stopping training after it reaches good performance on training data and then run prediction on same data gives same crappy result as validation pass during traing (or as model.eval()). This behavior makes transfer learning unusable. One cannot train model long after training converged for a chance that training and validation stats will eventually become similar. If it really still is BatchNormalization, it should be fixed and not hidden under carpet

Same model (as much as I could make it same) implemented with pytorch behaves as one would expect. Both training and validation stats are pretty much the same. No problems with prediction/evaluation of the model. Number of required epochs is reasonable (and similar to tensorflow training stats). Data fed into keras model and pytorch model were processed in almost exactly same way.

I guess I'll have to run my own training loop for now and update only layers added on top of mobilenet (and hope it will help). Is there some workaround except _just let it run wor 500 more epochs_?

@dmus how is this not a problem? How are you training your models?

**Comment by cecabert** [comment]
I've noticed similar behavior as @ccg1415 with a  pre-trained ResNet, the validation loss skyrocket through the roof

**Comment by ymodak** [comment]
Closing this issue for now since the issue author is aligned. However feel free to post a new issue if still have problems. Thanks you.
